src/lib/db/client.ts

```diff
@@ -0,0 +1,419 @@
+  return database.connection?.listRoutines(filter);
+}
+
+async function listTableColumns(server: IDbConnectionServer, database: IDbConnectionDatabase, table?: string, schema?: string) {
+  checkIsConnected(server , database);
+  return await database.connection?.listTableColumns(database.database, table, schema) || [];
+}
+
+function listMaterializedViewColumns(server: IDbConnectionServer, database: IDbConnectionDatabase, table: string, schema: string) {
+  checkIsConnected(server , database);
+  if (database.connection?.listMaterializedViewColumns) {
+    return database.connection?.listMaterializedViewColumns(database.database, table, schema)
+  } else {
+    return []
+  }
+}
+
+function listTableTriggers(server: IDbConnectionServer, database: IDbConnectionDatabase, table: string, schema: string) {
+  checkIsConnected(server , database);
+  return database.connection?.listTableTriggers(table, schema);
+}
+
+function listTableIndexes(server: IDbConnectionServer, database: IDbConnectionDatabase, table: string, schema: string) {
+  checkIsConnected(server , database);
+  return database.connection?.listTableIndexes(database.database, table, schema);
+}
+
+function getTableReferences(server: IDbConnectionServer, database: IDbConnectionDatabase, table: string, schema: string) {
+  checkIsConnected(server , database);
+  return database.connection?.getTableReferences(table, schema);
+}
+
+function getPrimaryKey(server: IDbConnectionServer, database: IDbConnectionDatabase, table: string, schema: string) {
+  checkIsConnected(server, database)
+  return database.connection?.getPrimaryKey(database.database, table, schema)
+}
+
+function getTableKeys(server: IDbConnectionServer, database: IDbConnectionDatabase, table: string, schema: string) {
+  checkIsConnected(server , database);
+  return database.connection?.getTableKeys(database.database, table, schema);
+}
+
+function query(server: IDbConnectionServer, database: IDbConnectionDatabase, queryText: string) {
+  checkIsConnected(server , database);
+  return database.connection?.query(queryText);
+}
+
+function updateValues(server: IDbConnectionServer, database: IDbConnectionDatabase, updates: IDbUpdate[]) {
+  checkIsConnected(server, database)
+  return database.connection?.updateValues(updates)
+}
+
+function executeQuery(server: IDbConnectionServer, database: IDbConnectionDatabase, queryText: string) {
+  checkIsConnected(server , database);
+  return database.connection?.executeQuery(queryText);
+}
+
+
+function listDatabases(server: IDbConnectionServer, database: IDbConnectionDatabase, filter: IDbFilter) {
+  checkIsConnected(server , database);
+  return database.connection?.listDatabases(filter);
+}
+
+
+async function getQuerySelectTop(server: IDbConnectionServer, database: IDbConnectionDatabase, table: string, schema: string, limit: number) {
+  checkIsConnected(server , database);
+  let limitValue = limit;
+  // TODO: (geovannimp) _limit is not declared
```

Comment:
I don't use this function, it's a leftover, so this is fine

Reply:
IE - don't worry about this function at all. You could just remove it if you like.